src/meta/server_state.cpp

```diff
@@ -1259,6 +1259,102 @@ void server_state::drop_app(dsn::message_ex *msg)
     }
 }
 
+void server_state::rename_app_unlock(configuration_rename_app_rpc rpc)
+{
+    zauto_write_lock l(_lock);
```

Comment:
Since the function requires the lock itself, the function name should be `do_rename_app` (or sth like that).

Further more, you can remove the lock if you pass `rpc`, `app_id`, and call `_exist_apps.emplace(...)` in rename_app?

Reply:
I see. I think merge two functions into one is better.